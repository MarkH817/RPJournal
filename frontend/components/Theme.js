import React from 'react'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import 'modern-normalize'
import 'nprogress/nprogress.css'
import '@reach/skip-nav/styles.css'
import 'draft-js/dist/Draft.css'
import 'draftail/dist/draftail.css'
import 'typeface-open-sans'
import 'typeface-sarabun'

const theme = {
  white: 'rgb(255,255,255)',
  offWhite: 'rgb(225,225,225)',
  lightGray: 'rgb(175,175,175)',
  gray: 'rgb(125,125,125)',
  darkGray: 'rgb(75,75,75)',
  black: 'rgb(25,25,25)',
  primaryColor: '#ea45ba',
  accentColor: '#1d5e81',
  headerFont: `'Sarabun', 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif`,
  baseFont: `'Open Sans', 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif`
}

const GlobalStyles = createGlobalStyle`
  * {
    outline-width: 2px;
  }

  body {
    color: ${props => props.theme.black};
    font-family: ${props => props.theme.baseFont};
    font-size: 1em;
    padding: 0;
  }

  a {
    color: ${props => props.theme.primaryColor};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${props => props.theme.headerFont};
  }

  .detail-notes {
    line-height: 1.5;
  }
`

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    <React.Fragment>
      <GlobalStyles />

      {children}
    </React.Fragment>
  </ThemeProvider>
)

export default Theme
