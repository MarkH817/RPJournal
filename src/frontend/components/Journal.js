import React from 'react'
import styled from 'styled-components'

import { fillerImage } from '../lib/filler'

const StyledJournal = styled.article`
  img {
    object-fit: contain;
    max-width: 100%;
  }
`

/**
 * @param {object} props
 * @param {JournalModel} props.journal
 * @returns {React.ReactElement}
 */
const Journal = props => {
  const { journal } = props

  const totalQuests = journal.quests.length
  const completedQuests = journal.quests.reduce(
    (prev, curr) => (curr.completed ? prev + 1 : prev),
    0
  )

  const img = fillerImage({ text: journal.title })

  return (
    <StyledJournal>
      <h1>{journal.title}</h1>

      <img src={img} alt={journal.title} />

      <p>{journal.description}</p>

      <footer>
        <p>
          Quests Completed: ({completedQuests}/{totalQuests})
        </p>
      </footer>
    </StyledJournal>
  )
}

export default Journal
