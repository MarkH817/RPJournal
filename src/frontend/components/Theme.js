import React from 'react'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import 'modern-normalize'
import 'nprogress/nprogress.css'
import '@reach/skip-nav/styles.css'
import 'draft-js/dist/Draft.css'
import 'draftail/dist/draftail.css'
import 'typeface-open-sans'
import 'typeface-sarabun'

const theme = {
  white: '#ffffff',
  offWhite: '#dfdfdf',
  lightGray: '#c3c3c3',
  gray: '#767676',
  darkGray: '#525252',
  black: '#353535',
  primaryColor: '#ea45ba',
  accentColor: '',
  headerFont: `'Sarabun', 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif`,
  baseFont: `'Open Sans', 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif`
}

const GlobalStyles = createGlobalStyle`
  body {
    color: ${props => props.theme.black};
    font-family: ${props => props.theme.baseFont};
    font-size: 1em;
    padding: 0;
  }

  a {
    color: ${props => props.theme.primaryColor};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${props => props.theme.headerFont};
  }
`

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    <React.Fragment>
      <GlobalStyles />

      {children}
    </React.Fragment>
  </ThemeProvider>
)

export default Theme
