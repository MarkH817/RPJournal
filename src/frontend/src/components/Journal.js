import React from 'react'
import dummy from 'dummyjs'
import styled from 'styled-components'

const StyledJournal = styled.article`
  img {
    object-fit: contain;
    max-height: 100%;
    max-width: 100%;
  }
`

/**
 * @typedef {object} Journal
 * @property {string} id
 * @property {string} title
 * @property {string} description
 * @property {number} entryCount
 */

/**
 * @param {object} props
 * @param {Journal} props.journal
 */
const Journal = props => {
  const { journal } = props

  return (
    <StyledJournal>
      <h1>{journal.title}</h1>

      <img src={dummy.image(640, 480)} alt={journal.title} />

      <p>{journal.description}</p>

      <footer>
        <p>Entry Count: {journal.entryCount}</p>
      </footer>
    </StyledJournal>
  )
}

export default Journal
