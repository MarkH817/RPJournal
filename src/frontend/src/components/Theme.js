import React from 'react'
import '@reach/skip-nav/styles.css'
import { createGlobalStyle, ThemeProvider } from 'styled-components'

const theme = {
  black: '#353535',
  primaryColor: 'royalblue'
}

const GlobalStyles = createGlobalStyle`
  *,
  *:after,
  *:before {
    box-sizing: border-box;
  }

  body {
    color: ${props => props.theme.black};
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    font-size: 1em;
    margin: 0;
    padding: 0;
  }

  a {
    color: ${props => props.theme.primaryColor};
  }
`

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    <React.Fragment>
      <GlobalStyles />

      {children}
    </React.Fragment>
  </ThemeProvider>
)

export default Theme
