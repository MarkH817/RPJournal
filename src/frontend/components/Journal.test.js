import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Journal from './Journal'
import Theme from './Theme'

/** @type {JournalModel} */
const exampleJournal = {
  id: '1',
  title: 'Example Journal',
  description: 'Lorem ipsum a et louve.',
  sessions: []
}

describe('Journal', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Theme>
          <Journal journal={exampleJournal} />
        </Theme>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
