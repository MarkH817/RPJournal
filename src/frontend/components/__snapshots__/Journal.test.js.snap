// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Journal renders correctly 1`] = `
.c0 img {
  object-fit: contain;
  max-width: 100%;
}

<article
  className="c0"
>
  <h1>
    Example Journal
  </h1>
  <img
    alt="Example Journal"
    src="https://via.placeholder.com/640x480.png/535353/eeeeee?text=Example%20Journal"
  />
  <p>
    Lorem ipsum a et louve.
  </p>
  <footer>
    <p>
      Quests Completed: (
      0
      /
      0
      )
    </p>
  </footer>
</article>
`;
